run-connector.js

```diff
@@ -1,33 +1,70 @@
+const { createApp } = require('ilp-connector')
+const { connectCoinCap } = require('@kava-labs/crypto-rate-utils')
+const chokidar = require('chokidar')
 
-const config = {
-	env: process.env.CONNECTOR_ENV,
-	adminApi: true,
-	adminApiPort: 7769,
-	ilpAddress: process.env.ILP_ADDRESS,
-	backend: 'ecb-plus-coinmarketcap',
-	spread: 0,
-	store: 'ilp-store-redis',
-	storeConfig: {
-		password: process.env.REDIS_PASS,
-		prefix: 'connector',
-		host: '127.0.0.1',
-		port: 6379
-	},
-	accounts: {
-		// Peers
-		'strata4-xrp': require('./plugins/strata4-xrp.config.js'),
-		// Servers
-		eth: require('./plugins/eth.config.js'),
-		xrp: require('./plugins/xrp.config.js'),
-		btc: require('./plugins/btc.config.js'),
-		// Moneyd
-		local: require('./plugins/mini-accounts.config.js')
-	}
-}
+async function run() {
+  const config = {
+    env: process.env.CONNECTOR_ENV,
+    adminApi: true,
+    adminApiPort: 7769,
+    ilpAddress: process.env.ILP_ADDRESS,
+    backend: '@kava-labs/ilp-backend-crypto',
+    spread: 0,
+    store: 'ilp-store-redis',
+    storeConfig: {
+      password: process.env.REDIS_PASS,
+      prefix: 'connector',
+      host: '127.0.0.1',
+      port: 6379
+    },
+    accounts: {}
+  }
+
+  const rateApi = await connectCoinCap()
+
+  const { listen, addPlugin, removePlugin } = createApp(config)
+
+  const setupWatcher = async accountId => {
+    const path = `./plugins/${accountId}.config.js`
+
+    // Setup a watcher for the file to hot swap the plugin if the config changes
+    const watcher = chokidar.watch(path, {
+      ignoreInitial: true
+    })
+
+    const load = async () => {
+      // Create the plugin options to pass to the connector
+      const createConfig = require(path)
+      const accountConfig = await createConfig(rateApi)
 
-if (process.env.NODE_ENV == 'development') {
-	console.log(`config ->`, config)
+      await addPlugin(accountId, accountConfig)
+    }
+
+    const reload = async () => {
+      await removePlugin(accountId)
+      await load()
+    }
+
+    watcher.on('ready', load)
+    watcher.on('change', reload)
+  }
+
+  // Start the conenctor
+  await listen()
+
+  const accountIds = [
```

Comment:
What about adding a peer or a new asym server without restarting? Could we maybe walk the `plugins` directory to recognize when new plugins are added?

Reply:
may be want to re-spell connector line 52? unless that was deliberate

Reply:
its funny when you can always find some one else's mistakes better than ones own :|